partiql-parser/src/main/antlr/PartiQLTokens.g4

```diff
@@ -388,12 +388,16 @@ LITERAL_STRING
     : '\'' ( ('\'\'') | ~('\'') )* '\'';
 
 LITERAL_INTEGER
-    : DIGIT DIGIT*;
+    : DIGIT+;
 
-LITERAL_DECIMAL:
-    DIGIT+ '.' DIGIT* ([e] [+-]? DIGIT+)?
-    | '.' DIGIT DIGIT* ([e] [+-]? DIGIT+)?
-    | DIGIT DIGIT* ([e] [+-]? DIGIT+)?
+LITERAL_DECIMAL
+    : DIGIT+ '.' DIGIT*
+    | '.' DIGIT+
+    ;
+
+LITERAL_FLOAT
+    : DIGIT+ ('.' DIGIT*)? 'E' [+-]? DIGIT+
+    | '.' DIGIT+ 'E' [+-]? DIGIT+
```

Comment:
I played around with some different ANTLR parsing modelings and it led to some other issues. I think the functionality is fine to parse as unsigned numerics. We could always change the modeling in the future or add some AstVisitor pass to unify unary +/- with AST numeric literals.

Changing literal parsing to be in the parser rather than lexer -- cannot use the single-quoted `'E'`, so `E` would need to be in the tokens.

Changing literal parsing in the tokens may bind too eagerly -- e.g. `1 - 23` would tokenize `-23` together.